eth/filters/filter_system_test.go

```diff
@@ -65,6 +65,13 @@ func (b *testBackend) HeaderByNumber(ctx context.Context, blockNr rpc.BlockNumbe
 			return nil, nil
 		}
 		num = *number
+	} else if blockNr == rpc.FinalizedBlockNumber {
```

Comment:
Shouldn't we add a clause for `SafeBlockNumber` too?

Reply:
Ideally yes, but since safe block isn't stored in the DB it'd require mocking it out in `testBackend`, which is definitely doable but I was worried about scope creep of the PR.